src/rgw/rgw_sal_motr.cc

```diff
@@ -3091,6 +3096,31 @@ int MotrStore::get_user_by_access_key(const DoutPrefixProvider *dpp, const std::
 
 int MotrStore::get_user_by_email(const DoutPrefixProvider *dpp, const std::string& email, optional_yield y, std::unique_ptr<User>* user)
 {
+  int rc;
+  User *u;
+  bufferlist bl;
+  RGWUserInfo uinfo;
+  MotrEmailInfo email_info; 
+  rc = do_idx_op_by_name(RGW_IAM_MOTR_EMAIL_KEY,
+                           M0_IC_GET, email, bl);
+  if (rc < 0){
+    ldout(cctx, 0) << "Email Id not found: rc = " << rc << dendl;
+    return rc;
+  }
+  auto iter = bl.cbegin();
+  email_info.decode(iter);
+  ldout(cctx, 0) << "Loading user: " << email_info.user_id << dendl;
+  uinfo.user_id.id = email_info.user_id;
```

Comment:
This will not append the tenant. Update as below:
`uinfo.user_id.from_str(email_info.user_id);`

Reply:
done